lib/io.toit

```diff
@@ -754,7 +808,9 @@ abstract class Reader implements old-reader.Reader:
   ```
   */
   big-endian -> EndianReader:
-    return EndianReader --reader=this --byte-order=binary.BIG_ENDIAN
+    if not endian_ or endian_.byte-order_ != binary.BIG_ENDIAN:
```

Comment:
I wouldn't read the field that many times.

Reply:
done.